include/highfive/H5Object.hpp

```diff
@@ -63,6 +63,11 @@ class Object {
     ///
     ObjectType getType() const;
 
+    // Check if refer to same object
+    bool operator==(const Object& other) const noexcept {
+        return _hid == other._hid;
+    }
+
```

Comment:
https://github.com/BlueBrain/HighFive/issues/401 https://forum.hdfgroup.org/t/h5l-info2-t-me-meaning-of-member-variables/8162/2?u=kerim.khemraev
objects with different hid can be the same.

Reply:
I'm not sure I understand the comment.. but does that mean one should call https://portal.hdfgroup.org/display/HDF5/H5I_OBJECT_VERIFY for `_hid` and `other._hid` and compare the addresses?

Reply:
I don't know, I'm asking what does it mean for user `Obj1 == Obj2` ?

Reply:
> objects with different hid can be the same.

But I wonder if we should say they are the same... Even if they point to the same thing in disk, I think in the general case it's good to compare the hid. If, in the case of datasets, we can do a better comparison, sure. But remember this definition is shared among any kind of hdf5 object.